### ISSUE TITLE
Enable VTI typechecking

### ISSUE THREAD

**Issue by luxaritas** [eternagame/eternagame.org#232]
See https://vuejs.github.io/vetur/guide/vti.html

There's a fair amount of errors this spits out, so I don't have time to go after this now. However, it should be a relatively straightforward to work through these errors

**Comment by snehansh** [comment]
@luxaritas

I see that you have added details like:
There's a fair amount of errors this spits out..

Are you referring to errors that gets displayed in VS code in the "PROBLEMS" section when we open a .vue page in VS code.

For example, please refer screenshot below:
![image](https://github.com/eternagame/eternagame.org/assets/35124383/89d0bb77-04f4-4440-85d7-67aa69bf34bc)

**Comment by luxaritas** [comment]
@snehansh Yes and no :) What I was referring to was actually running the `vti` command, but the errors it would raise are the same as what you should see in the problems section - it just checks the entire codebase (just like if you ran `tsc --noEmit` on a pure typescript codebase - you could open a file and see the errors for just that file, or run the command to get the commands across the entire codebase).

**Comment by snehansh** [comment]
@luxaritas 
I am interested in working on this issue.
Please assign this to me. Thanks.

**Comment by luxaritas** [comment]
Sure, thanks! Also noting as discussed in Discord that we probably want to use vue-tsc instead as it's newer (plus vti appears to have issues)
